module3/repository/repository.go

```diff
@@ -0,0 +1,98 @@
+package repository
+
+import (
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"sync"
+)
+
+type Repository struct {
+	mt   sync.RWMutex
+	data map[string]User
+}
+
+type User struct {
+	Uuid           uuid.UUID
+	UserName       string
+	HashedPassword string
```

Comment:
Uuid seems to be redundantly. This field does not make sense, because we don't use it. There is no according field in the original schema, nor any practical use in the code. The only place where we can use uuid is the response, but in that case it would be better to link the uuid to processing request (like request-id for the logging purposes), not to the data.